tracing/trace_state.cc

```diff
@@ -142,7 +142,7 @@ void trace_state::build_parameters_map() {
     params_values& vals = *_params_ptr;
 
     if (vals.batchlog_endpoints) {
-        params_map.emplace("batch_endpoints", join(sstring(","),  *vals.batchlog_endpoints | boost::adaptors::transformed([](gms::inet_address ep) {return seastar::format("/{}", ep);})));
+        params_map.emplace("batch_endpoints", fmt::to_string(*vals.batchlog_endpoints | boost::adaptors::transformed([](gms::inet_address ep) {return seastar::format("/{}", ep);})));
```

Comment:
If I understand correctly, this changes the format of this output, adding braces. Is this ok?
I'm not at all a fan of using fmt::to_string with some unspecified formatting for outputting strings which are NOT log messages (as I think is the case here) and where it's not clear that changing the format one day wouldn't break anything.

Can you please make sure that whatever this build_parameters_map() is doing, we have a test for it, so if the format was broken we would know it?

Reply:
I don't think it matters, but to be on the safe side, I'm just converting this to use fmt::join in v5 so there would be no change in the string format.